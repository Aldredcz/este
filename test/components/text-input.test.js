// @flow
import TextInput from '../../components/text-input';
import { createExpectRender } from './utils';

const theme = {
  typography: {
    fontSize: level => level * 2 + 16, // 0 = 16, 1 = 18, -1 = 14 etc.
    rhythm: ratio => ratio * 24,
    lineHeight: 24,
  },
  colors: {
    primary: 'blue',
    success: 'green',
    warning: 'orange',
    danger: 'red',
    black: 'black',
    white: 'white',
    gray: 'gray',
  },
  text: {
    bold: 600,
    color: 'black',
    // github.com/twbs/bootstrap/blob/v4-dev/scss/_variables.scss
    fontFamily: '-apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, "Helvetica Neue", Arial, sans-serif',
    rawStyle: {
      fixFontSmoothing: true,
    },
  },
  textInput: {
    disabledOpacity: 0.5,
  },
};

const expectRender = createExpectRender(theme);

test('render', () => {
  expectRender(() => <TextInput />);
});

test('disabled', () => {
  expectRender(() => <TextInput disabled />);
});

test('color', () => {
  expectRender(() => <TextInput color="primary" />);
});
