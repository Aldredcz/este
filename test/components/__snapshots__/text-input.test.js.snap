// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`color 1`] = `
Object {
  "component": <input
    className="a b c d e f g h i j k l m"
/>,
  "fela": "
    .a{display:inline}
    .b{flex-direction:column}
    .c{position:relative}
    .d{white-space:pre-wrap}
    .e{overflow-wrap:break-word}
    .f{-ms-hyphens:auto}
    .g{color:blue}
    .h{font-family:-apple-system, BlinkMacSystemFont, \\"Segoe UI\\", Roboto, \\"Helvetica Neue\\", Arial, sans-serif}
    .i{font-size:16}
    .j{line-height:24px}
    .k::placeholder{color:rgba(0, 0, 255, 0.5)}
    .l{background-color:transparent}
    .m{outline:none}",
}
`;

exports[`disabled 1`] = `
Object {
  "component": <input
    className="a b c d e f g h i j k l m n"
    disabled={true}
/>,
  "fela": "
    .a{display:inline}
    .b{flex-direction:column}
    .c{position:relative}
    .d{opacity:0.5}
    .e{white-space:pre-wrap}
    .f{overflow-wrap:break-word}
    .g{-ms-hyphens:auto}
    .h{color:black}
    .i{font-family:-apple-system, BlinkMacSystemFont, \\"Segoe UI\\", Roboto, \\"Helvetica Neue\\", Arial, sans-serif}
    .j{font-size:16}
    .k{line-height:24px}
    .l::placeholder{color:rgba(0, 0, 0, 0.5)}
    .m{background-color:transparent}
    .n{outline:none}",
}
`;

exports[`render 1`] = `
Object {
  "component": <input
    className="a b c d e f g h i j k l m"
/>,
  "fela": "
    .a{display:inline}
    .b{flex-direction:column}
    .c{position:relative}
    .d{white-space:pre-wrap}
    .e{overflow-wrap:break-word}
    .f{-ms-hyphens:auto}
    .g{color:black}
    .h{font-family:-apple-system, BlinkMacSystemFont, \\"Segoe UI\\", Roboto, \\"Helvetica Neue\\", Arial, sans-serif}
    .i{font-size:16}
    .j{line-height:24px}
    .k::placeholder{color:rgba(0, 0, 0, 0.5)}
    .l{background-color:transparent}
    .m{outline:none}",
}
`;
